s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -148,7 +149,11 @@ func (t *TidbClient) GetBuckets(ctx context.Context) (buckets []*Bucket, err err
 	sqltext := "select bucketname,tenantid,userid,createtime,usages,location,deleted,acl,cors,lc,policy," +
 		"versioning,replication,update_time,tiers from buckets;"
 
-	if !isAdmin {
+	if filterLoc != "" {
```

Comment:
I think this part can be improved, to handle any other filteroptions comes

Reply:
improved

Reply:
when this sql text is just concatenated with condition instead of reassigning it is giving its giving 'error 1105 i.e exceeding length of query'